api-gateway/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Base stage for both dev and prod
+FROM node:18-alpine AS base
+
+# Install pnpm globally
+RUN npm install -g pnpm
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Copy package.json and pnpm-lock.yaml
+COPY package.json ./
+COPY pnpm-lock.yaml ./
+
+RUN pnpm install
+
+# Development stage
+FROM base AS development
+
+COPY src ./src
+COPY tsconfig.json ./
+COPY .env.dev ./
```

Comment:
I think we dont need to copy in the .env.dev as it is referenced in the compose.yml